### PR TITLE
Prevent creation of bed with the same name

### PR DESCRIPTION
**Pull request by iamsdas**
Prevent the creation of a bed, if a bed with the same name exists in that location.
Fixes: [#2282](https://github.com/coronasafe/care_fe/issues/2282)

### CODE REVIEW

**Review by vigneshhari — CHANGES_REQUESTED** [review]
This validation should be written in the validate method, these validations can also be written at the model level which might make more sense.

**Comment by iamsdas** [comment]
@vigneshhari
I updated the PR to add the validations in the model level (using the `clean` method) and made some minor changes so that DRF can handle these Django validation errors.

File: care/facility/models/bed.py [hunk 1]
```diff
@@ -35,6 +36,14 @@ class BedType(enum.Enum):
     assets = models.ManyToManyField(Asset, through="AssetBed")
     location = models.ForeignKey(AssetLocation, on_delete=models.PROTECT, null=False, blank=False)
 
+    def clean(self) -> None:
+        if Bed.objects.filter(location=self.location, name=self.name).exists():
+            raise ValidationError({"name": "Bed with same name already exists in location."})
+
+    def save(self, *args, **kwargs) -> None:
+        self.clean()
```
Comment (vigneshhari) [inline care/facility/models/bed.py:44]: validate makes more sense here since you are not really cleaning anything.

**Review by vigneshhari — APPROVED** [review]

**Comment by vigneshhari** [comment]
resolve the conflicts and we are good to merge.

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**